### ISSUE TITLE
Could it be used for binary classification

### ISSUE THREAD

**Issue by yszd577** [Extreme-classification/deepxml#12]
Hey!
you really did a great job! I would like to use it for binary classification tasks, and after I read the code, I change the config json below:
```json
{
    "global": {
        **"dataset": "IMDB",** 
        "feature_type": "sparse",
        **"num_labels": 2,**
        "arch": "Astec",
        "A": 0.55,
        "B": 1.5,
        "use_reranker": false,
        "surrogate_threshold": 1024,
        **"surrogate_method": 0,**
        "embedding_dims": 300,
        **"top_k": 2,**
	    "beta": 0.3,
        "save_predictions": true,
        "trn_label_fname": "trn_X_Y.txt",
        "val_label_fname": "tst_X_Y.txt",
        "tst_label_fname": "tst_X_Y.txt",
        "trn_feat_fname": "trn_X_Xf.txt",
        "val_feat_fname": "tst_X_Xf.txt",
        "tst_feat_fname": "tst_X_Xf.txt"
    },
    "surrogate": {
        "num_epochs": 20,
        "dlr_factor": 0.5,
        "learning_rate": 0.01,
        "batch_size": 255,
        "dlr_step": 14,
        "normalize": true,
        "init": null,
        "optim": "Adam",
        "embeddings": "fasttextB_embeddings_300d.npy",
        "validate": true,
        "save_intermediate": true
    },
    "extreme": {
        "num_epochs": 20,
        "dlr_factor": 0.5,
        "learning_rate": 0.007,
        "batch_size": 255,
        "dlr_step": 14,
        "ns_method": "ensemble",
        "num_centroids": 1,
        "efC": 300,
        "efS": 400,
        "M": 100,
        "num_nbrs": 500,
        "ann_threads": 18,
        "beta": 0.5,
        "surrogate_mapping": null,
        "num_clf_partitions": 1,
        "optim": "Adam",
        "freeze_intermediate": true,
        "validate": true,
        **"model_method": "full",**
        "normalize": true,
        "shortlist_method": "hybrid",
        "init": null,
        "use_shortlist": false,
        "use_intermediate_for_shorty": true
    },
    "reranker": {
        "num_epochs": 15,
        "dlr_factor": 0.5,
        "learning_rate": 0.005,
        "batch_size": 255,
        "dlr_step": 10,
        "beta": 0.6,
        "num_clf_partitions": 1,
        "optim": "Adam",
        "validate": true,
        "model_method": "reranker",
        "shortlist_method": "static",
        "surrogate_mapping": null,
        "normalize": true,
        "use_shortlist": true,
        "init": null,
        "save_intermediate": false,
        "keep_invalid": true,
        "freeze_intermediate": false,
        "update_shortlist": false,
        "use_pretrained_shortlist": true,
        "embeddings": "fasttextB_embeddings_300d.npy"
    }
}
```
and I change items with bold, cause it seems label clusters and shortlists can't work well in such situation, so I block them. When I get the tst_predictions.npz file, I found it's all 0 or 1 after sigmoid or softmax(I don't konw how you handle that) , So my seeting is wrong? or Astec couldn't do binary classification ? 
Thanks!

**Comment by kunaldahiya** [comment]
The pipeline with shortlist etc. is designed for large output space in mind. 'DeepXML-OVA' would be more suitable for a binary classification task. 

Example usage (please add/change config file in the configs/DeepXML-OVA directory): `./run_main.sh 0 DeepXML-OVA EURLex-4K 0 108`

**Comment by yszd577** [comment]
I made it.Thank you!
